hoodie-client/src/main/java/com/uber/hoodie/HoodieWriteClient.java

```diff
@@ -1051,10 +1052,16 @@ public static SparkConf registerClasses(SparkConf conf) {
   /**
    * Deduplicate Hoodie records, using the given deduplication funciton.
    */
-  private JavaRDD<HoodieRecord<T>> deduplicateRecords(JavaRDD<HoodieRecord<T>> records,
+  JavaRDD<HoodieRecord<T>> deduplicateRecords(JavaRDD<HoodieRecord<T>> records,
       int parallelism) {
     return records
-        .mapToPair(record -> new Tuple2<>(record.getKey(), record))
+        .mapToPair(record -> {
+          HoodieKey hoodieKey = record.getKey();
+          // If index used is global, then records are expected to differ in their partitionPath
+          boolean isGlobal = getIndex() != null && getIndex().isGlobal();
```

Comment:
why would the index be null?

Reply:
Corrected this. The check had to be outside of this code block.